jetty-websocket/javax-websocket-server/src/main/java/org/eclipse/jetty/websocket/javax/server/JavaxWebSocketServerContainer.java

```diff
@@ -88,26 +87,39 @@ public static JavaxWebSocketServerContainer ensureContainer(ServletContext servl
         JavaxWebSocketServerContainer container = contextHandler.getBean(JavaxWebSocketServerContainer.class);
         if (container==null)
         {
-            // Find Pre-Existing (Shared?) HttpClient and/or executor
-            HttpClient httpClient = (HttpClient)servletContext.getAttribute(JavaxWebSocketServletContainerInitializer.HTTPCLIENT_ATTRIBUTE);
-            if (httpClient == null)
-                httpClient = (HttpClient)contextHandler.getServer()
-                    .getAttribute(JavaxWebSocketServletContainerInitializer.HTTPCLIENT_ATTRIBUTE);
-
-            Executor executor = httpClient == null?null:httpClient.getExecutor();
-            if (executor == null)
-                executor = (Executor)servletContext.getAttribute("org.eclipse.jetty.server.Executor");
-            if (executor == null)
-                executor = contextHandler.getServer().getThreadPool();
-
-            if (httpClient != null && httpClient.getExecutor() == null)
-                httpClient.setExecutor(executor);
+            Supplier<WebSocketCoreClient> coreClientSupplier = () ->
+            {
+                WebSocketCoreClient coreClient = contextHandler.getBean(WebSocketCoreClient.class);
+                if (coreClient == null)
+                {
+                    // Find Pre-Existing (Shared?) HttpClient and/or executor
+                    HttpClient httpClient = (HttpClient)servletContext.getAttribute(JavaxWebSocketServletContainerInitializer.HTTPCLIENT_ATTRIBUTE);
+                    if (httpClient == null)
+                        httpClient = (HttpClient)contextHandler.getServer().getAttribute(JavaxWebSocketServletContainerInitializer.HTTPCLIENT_ATTRIBUTE);
+
+                    Executor executor = httpClient == null?null:httpClient.getExecutor();
+                    if (executor == null)
+                        executor = (Executor)servletContext.getAttribute("org.eclipse.jetty.server.Executor");
+                    if (executor == null)
+                        executor = contextHandler.getServer().getThreadPool();
+
+                    if (httpClient != null && httpClient.getExecutor() == null)
+                        httpClient.setExecutor(executor);
+
+                    // create the core client
+                    coreClient = new WebSocketCoreClient(httpClient, WebSocketComponents.ensureWebSocketComponents(servletContext));
+                    coreClient.getHttpClient().setName("Javax-WebSocketClient@" + Integer.toHexString(coreClient.getHttpClient().hashCode()));
+                    if (executor != null && httpClient == null)
+                        coreClient.getHttpClient().setExecutor(executor);
```

Comment:
Do we need this?  isn't this done by 106-107 above?

Reply:
line 106-107 sets the executor on the HttpClient if the found HttpClient does not have an executor

line 112-113 sets the executor on the CoreClients newly created HttpClient because we didn't find a HttpClient therefore it was created by the CoreClient

Reply:
Gotcha